### ISSUE TITLE
wlcs tests flaky in CI

### ISSUE THREAD

**Issue by Saviq** [MirServer/mir#2474]
These failures are on arm{hf,64}, which run through qemu, so that might be a factor.

E.g.

https://github.com/MirServer/mir/runs/6879373164?check_suite_focus=true#step:5:9640

```
[ RUN      ] AllSurfaceTypes/TouchTest.sends_touch_up_on_surface_destroy/subsurface_at_x7_y12
[2022-06-14 11:49:25.832214] <information> mirserver: Starting
[2022-06-14 11:49:25.842273] < - debug - > mirserver: No session management supported
[2022-06-14 11:49:25.845956] <information> VT switch key handler: No VT switching support available: MinimalConsoleServices does not support VT switching
[2022-06-14 11:49:25.922347] <information> mircommon: Loading modules from: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/
[2022-06-14 11:49:25.923540] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-wayland.so.20
[2022-06-14 11:49:25.923894] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-eglstream-kms.so.20
[2022-06-14 11:49:25.924024] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-gbm-kms.so.20
[2022-06-14 11:49:25.924116] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/server-x11.so.20
[2022-06-14 11:49:25.924200] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/input-evdev.so.8
[2022-06-14 11:49:25.924284] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-dummy.so
[2022-06-14 11:49:25.924364] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/input-stub.so
[2022-06-14 11:49:25.924441] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-throw.so
[2022-06-14 11:49:25.930734] <information> mirserver: Found display driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 11:49:25.936042] <information> mirserver: Driver supports:
[2022-06-14 11:49:25.936740] <information> mirserver: 	System (priority 1)
[2022-06-14 11:49:25.939756] < -warning- > mirserver: Manually-specified display platform mir:stub-graphics does not claim to support this system. Trying anyway...
[2022-06-14 11:49:25.941784] <information> mirserver: Selected display driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 11:49:25.956756] <information> mircommon: Loading modules from: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/
[2022-06-14 11:49:25.957643] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-wayland.so.20
[2022-06-14 11:49:25.957903] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-eglstream-kms.so.20
[2022-06-14 11:49:25.958053] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-gbm-kms.so.20
[2022-06-14 11:49:25.958159] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/server-x11.so.20
[2022-06-14 11:49:25.958252] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/input-evdev.so.8
[2022-06-14 11:49:25.958349] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-dummy.so
[2022-06-14 11:49:25.958440] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/input-stub.so
[2022-06-14 11:49:25.958525] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-throw.so
[2022-06-14 11:49:25.959758] <information> mirserver: Found rendering driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 11:49:25.960062] <information> mirserver: Driver supports:
[2022-06-14 11:49:25.960151] <information> mirserver: 	System (priority 1)
[2022-06-14 11:49:25.960544] < -warning- > mirserver: Manually-specified rendering platform mir:stub-graphics does not claim to support this system. Trying anyway...
[2022-06-14 11:49:25.961105] <information> mirserver: Selected rendering driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 11:49:26.095698] <information> mirserver: Using software cursor
[2022-06-14 11:49:26.274314] < - ERROR - > : failed to create BasicScreenShooter: /<<PKGBUILDDIR>>/src/renderers/gl/renderer.cpp(232): Throw in function static GLuint mir::renderer::gl::Renderer::ProgramFactory::compile_shader(GLenum, const GLchar*)
Dynamic exception type: boost::wrapexcept<std::system_error>
std::exception::what: Failed to create shader: GL_NO_ERROR (0)
[2022-06-14 11:49:26.381550] <information> mirserver: Selected input driver: mir:stub-input (version: 2.8.0)
[2022-06-14 11:49:26.409426] <information> mirserver: Mir version 2.8.0
[2022-06-14 11:49:26.532430] <information> mirserver: Initial display configuration:
[2022-06-14 11:49:26.534459] <information> mirserver: * Output 1: VGA connected, used
[2022-06-14 11:49:26.536588] <information> mirserver: . |_ Physical size 0.0" 0x0mm
[2022-06-14 11:49:26.536722] <information> mirserver: . |_ Power is on
[2022-06-14 11:49:26.537784] <information> mirserver: . |_ Current mode 1600x1600 60.00Hz
[2022-06-14 11:49:26.537969] <information> mirserver: . |_ Preferred mode 1600x1600 60.00Hz
[2022-06-14 11:49:26.538082] <information> mirserver: . |_ Orientation normal
[2022-06-14 11:49:26.539098] <information> mirserver: . |_ Logical size 1600x1600
[2022-06-14 11:49:26.539367] <information> mirserver: . |_ Logical position +0+0
[2022-06-14 11:49:26.539468] <information> mirserver: . |_ Scaling factor: 1.00
*** stack smashing detected ***: terminated
Mir fatal error: Unsupported attempt to continue after a fatal signal: SIGABRT
qemu: uncaught target signal 6 (Aborted) - core dumped
```

**Comment by Saviq** [comment]
Similar:

https://github.com/MirServer/mir/runs/6879373239?check_suite_focus=true#step:5:8389

```plain
[ RUN      ] BadBufferTest.test_truncated_shm_file
[2022-06-14 11:48:49.554785] <information> mirserver: Starting
[2022-06-14 11:48:49.568380] < - debug - > mirserver: No session management supported
[2022-06-14 11:48:49.572312] <information> VT switch key handler: No VT switching support available: MinimalConsoleServices does not support VT switching
[2022-06-14 11:48:49.687369] <information> mircommon: Loading modules from: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/
[2022-06-14 11:48:49.688311] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/graphics-wayland.so.20
[2022-06-14 11:48:49.688622] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/graphics-eglstream-kms.so.20
[2022-06-14 11:48:49.688788] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/graphics-gbm-kms.so.20
[2022-06-14 11:48:49.688923] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/server-x11.so.20
[2022-06-14 11:48:49.689043] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/input-evdev.so.8
[2022-06-14 11:48:49.689172] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/graphics-dummy.so
[2022-06-14 11:48:49.689298] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/input-stub.so
[2022-06-14 11:48:49.690812] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/graphics-throw.so
[2022-06-14 11:48:49.697089] <information> mirserver: Found display driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 11:48:49.702391] <information> mirserver: Driver supports:
[2022-06-14 11:48:49.703025] <information> mirserver: 	System (priority 1)
[2022-06-14 11:48:49.705888] < -warning- > mirserver: Manually-specified display platform mir:stub-graphics does not claim to support this system. Trying anyway...
[2022-06-14 11:48:49.708045] <information> mirserver: Selected display driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 11:48:49.725770] <information> mircommon: Loading modules from: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/
[2022-06-14 11:48:49.726436] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/graphics-wayland.so.20
[2022-06-14 11:48:49.726627] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/graphics-eglstream-kms.so.20
[2022-06-14 11:48:49.726768] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/graphics-gbm-kms.so.20
[2022-06-14 11:48:49.726914] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/server-x11.so.20
[2022-06-14 11:48:49.727031] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/input-evdev.so.8
[2022-06-14 11:48:49.727157] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/graphics-dummy.so
[2022-06-14 11:48:49.727282] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/input-stub.so
[2022-06-14 11:48:49.728813] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-armhf/lib/server-modules/graphics-throw.so
[2022-06-14 11:48:49.730006] <information> mirserver: Found rendering driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 11:48:49.730291] <information> mirserver: Driver supports:
[2022-06-14 11:48:49.730385] <information> mirserver: 	System (priority 1)
[2022-06-14 11:48:49.730811] < -warning- > mirserver: Manually-specified rendering platform mir:stub-graphics does not claim to support this system. Trying anyway...
[2022-06-14 11:48:49.731348] <information> mirserver: Selected rendering driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 11:48:49.880123] <information> mirserver: Using software cursor
[2022-06-14 11:48:50.078135] < - ERROR - > : failed to create BasicScreenShooter: /<<PKGBUILDDIR>>/src/renderers/gl/renderer.cpp(232): Throw in function static GLuint mir::renderer::gl::Renderer::ProgramFactory::compile_shader(GLenum, const GLchar*)
Dynamic exception type: boost::wrapexcept<std::system_error>
std::exception::what: Failed to create shader: GL_NO_ERROR (0)
[2022-06-14 11:48:50.199543] <information> mirserver: Selected input driver: mir:stub-input (version: 2.8.0)
[2022-06-14 11:48:50.230327] <information> mirserver: Mir version 2.8.0
[2022-06-14 11:48:50.360791] <information> mirserver: Initial display configuration:
[2022-06-14 11:48:50.361601] <information> mirserver: * Output 1: VGA connected, used
[2022-06-14 11:48:50.363299] <information> mirserver: . |_ Physical size 0.0" 0x0mm
[2022-06-14 11:48:50.363422] <information> mirserver: . |_ Power is on
[2022-06-14 11:48:50.364504] <information> mirserver: . |_ Current mode 1600x1600 60.00Hz
[2022-06-14 11:48:50.364678] <information> mirserver: . |_ Preferred mode 1600x1600 60.00Hz
[2022-06-14 11:48:50.364782] <information> mirserver: . |_ Orientation normal
[2022-06-14 11:48:50.365701] <information> mirserver: . |_ Logical size 1600x1600
[2022-06-14 11:48:50.365942] <information> mirserver: . |_ Logical position +0+0
[2022-06-14 11:48:50.366066] <information> mirserver: . |_ Scaling factor: 1.00
Mir fatal error: Unsupported attempt to continue after a fatal signal: SIGSEGV
Mir fatal error: Unsupported attempt to continue after a fatal signal: SIGABRT
qemu: uncaught target signal 6 (Aborted) - core dumped
```

**Comment by Saviq** [comment]
But this one is native:

https://launchpadlibrarian.net/607067476/buildlog_ubuntu-focal-arm64.mir_2.8.0+dev225-gd7826d3c16-0ubuntu20.04_BUILDING.txt.gz

```plain
[ RUN      ] MultiRectEdges/RegionSurfaceInputCombinations.input_inside_region_seen/58
[2022-06-14 13:35:49.194786] <information> mirserver: Starting
[2022-06-14 13:35:49.194845] < - debug - > mirserver: No session management supported
[2022-06-14 13:35:49.194977] <information> VT switch key handler: No VT switching support available: MinimalConsoleServices does not support VT switching
[2022-06-14 13:35:49.195240] <information> mircommon: Loading modules from: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/
[2022-06-14 13:35:49.195449] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-eglstream-kms.so.20
[2022-06-14 13:35:49.195546] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-gbm-kms.so.20
[2022-06-14 13:35:49.195612] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/server-x11.so.20
[2022-06-14 13:35:49.195719] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-wayland.so.20
[2022-06-14 13:35:49.195800] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/input-evdev.so.8
[2022-06-14 13:35:49.197809] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-dummy.so
[2022-06-14 13:35:49.197914] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-throw.so
[2022-06-14 13:35:49.197968] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/input-stub.so
[2022-06-14 13:35:49.198172] <information> mirserver: Found display driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 13:35:49.198263] <information> mirserver: Driver supports:
[2022-06-14 13:35:49.198317] <information> mirserver: 	System (priority 1)
[2022-06-14 13:35:49.198367] < -warning- > mirserver: Manually-specified display platform mir:stub-graphics does not claim to support this system. Trying anyway...
[2022-06-14 13:35:49.198444] <information> mirserver: Selected display driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 13:35:49.198825] <information> mircommon: Loading modules from: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/
[2022-06-14 13:35:49.199049] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-eglstream-kms.so.20
[2022-06-14 13:35:49.199149] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-gbm-kms.so.20
[2022-06-14 13:35:49.199238] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/server-x11.so.20
[2022-06-14 13:35:49.199301] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-wayland.so.20
[2022-06-14 13:35:49.199357] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/input-evdev.so.8
[2022-06-14 13:35:49.201421] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-dummy.so
[2022-06-14 13:35:49.201538] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/graphics-throw.so
[2022-06-14 13:35:49.201623] <information> mircommon: Loading module: /<<PKGBUILDDIR>>/build-arm64/lib/server-modules/input-stub.so
[2022-06-14 13:35:49.201808] <information> mirserver: Found rendering driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 13:35:49.201891] <information> mirserver: Driver supports:
[2022-06-14 13:35:49.201942] <information> mirserver: 	System (priority 1)
[2022-06-14 13:35:49.201997] < -warning- > mirserver: Manually-specified rendering platform mir:stub-graphics does not claim to support this system. Trying anyway...
[2022-06-14 13:35:49.202081] <information> mirserver: Selected rendering driver: mir:stub-graphics (version 2.8.0)
[2022-06-14 13:35:49.202723] <information> mirserver: Using software cursor
[2022-06-14 13:35:49.217648] < - ERROR - > : failed to create BasicScreenShooter: /<<PKGBUILDDIR>>/src/renderers/gl/renderer.cpp(232): Throw in function static GLuint mir::renderer::gl::Renderer::ProgramFactory::compile_shader(GLenum, const GLchar*)
Dynamic exception type: boost::wrapexcept<std::system_error>
std::exception::what: Failed to create shader: GL_NO_ERROR (0)

[2022-06-14 13:35:49.218421] <information> mirserver: Selected input driver: mir:stub-input (version: 2.8.0)
[2022-06-14 13:35:49.218612] <information> mirserver: Mir version 2.8.0
[2022-06-14 13:35:49.219424] <information> mirserver: Initial display configuration:
[2022-06-14 13:35:49.219493] <information> mirserver: * Output 1: VGA connected, used
[2022-06-14 13:35:49.219518] <information> mirserver: . |_ Physical size 0.0" 0x0mm
[2022-06-14 13:35:49.219537] <information> mirserver: . |_ Power is on
[2022-06-14 13:35:49.219579] <information> mirserver: . |_ Current mode 1600x1600 60.00Hz
[2022-06-14 13:35:49.219618] <information> mirserver: . |_ Preferred mode 1600x1600 60.00Hz
[2022-06-14 13:35:49.219685] <information> mirserver: . |_ Orientation normal
[2022-06-14 13:35:49.219724] <information> mirserver: . |_ Logical size 1600x1600
[2022-06-14 13:35:49.219763] <information> mirserver: . |_ Logical position +0+0
[2022-06-14 13:35:49.219800] <information> mirserver: . |_ Scaling factor: 1.00
[2022-06-14 13:35:49.234039] <information> mirserver: Stopping
malloc(): unsorted double linked list corrupted
```

**Comment by Saviq** [comment]
Fixed by #2490
